libcaf_core/caf/detail/ring_buffer.test.cpp

```diff
@@ -0,0 +1,140 @@
+
+using namespace caf;
+
+using detail::ring_buffer;
+
+TEST("push_back adds element") {
+  ring_buffer<int> buf{3};
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 3; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("additional element after full capacity");
+  buf.push_back(4);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 2);
+  buf.push_back(5);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.push_back(6);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 4);
+}
+
+TEST("pop_front removes element") {
+  ring_buffer<int> buf{3};
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 3; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("remove element from buffer");
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 2);
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.pop_front();
+  check_eq(buf.empty(), true);
+}
+
+TEST("circular buffer overwrites oldest element after it is full") {
+  ring_buffer<int> buf{5};
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 5; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("add some elements into buffer");
+  buf.push_back(6);
+  buf.push_back(7);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.push_back(8);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 4);
+  buf.push_back(9);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 5);
+}
+
+TEST("new elements overwrite old elements that are popped") {
+  ring_buffer<int> buf{5};
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 5; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("remove some element from buffer");
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 2);
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.front(), 3);
+  info("add some elements into buffer");
+  buf.push_back(6);
+  buf.push_back(7);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.push_back(8);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 4);
+  buf.push_back(9);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 5);
+}
+
+TEST("empty ring buffer ignores call to push_back") {
```

Comment:
If empty ring buffer ignored `push_back`, we had a problem. Did you mean buffers with capacity 0?

```suggestion
TEST("push_back does nothing for ring buffer with a capacity of 0") {
```